packages/apostrophe/modules/@apostrophecms/rich-text-widget/ui/apos/components/AposRichTextWidgetEditor.vue

```diff
@@ -329,6 +339,31 @@ export default {
     }
   },
   watch: {
+    modelValue(newVal, oldVal) {
+      const [ newValStyles, oldValStyles ] = Object.entries(newVal)
+        .reduce((acc, [ fieldName, value ]) => {
+          const styleFields = this.moduleOptions.fieldsGroupStyle?.fields || [];
+          if (!styleFields.includes(fieldName)) {
+            return acc;
+          }
+          return [
+            {
+              ...acc[0],
+              [fieldName]: value
+            },
+            {
+              ...acc[1],
+              [fieldName]: oldVal[fieldName]
+            }
+          ];
+        }, [ {}, {} ]);
+
+      if (isEqual(newValStyles, oldValStyles)) {
+        return;
+      }
```

Comment:
This logic might be overkill for rich-text (but it has no cost). Recomputing styles only if they changes, nothing if content or other stuff changes like links etc.